model_analyzer/analyzer.py

```diff
@@ -206,9 +206,8 @@ def _should_profile_multiple_models_concurrently(self):
         return False
 
     def _get_profile_complete_string(self):
-        profiled_model_list = list(
-            self._state_manager.get_state_variable(
-                'ResultManager.results').keys())
+        profiled_model_list = self._state_manager.get_state_variable(
```

Comment:
I was hoping that all of this `_state_manager.get_state_variable('ResultManager.results')` stuff was going to go away, or at least only stored/loaded once. Can we not pull the data into a local instance of your class and operate directly on it?

Reply:
TMA-542 created